Explorer/assignment-2/src/pages/DetailsPage/DetailsPage.js

```diff
@@ -0,0 +1,96 @@
+import Header from "../../components/Header/Header";
+import "./DetailsPage.css";
+import { useState, useEffect } from "react";
+import Footer from "../../components/Footer/Footer";
+import { useParams } from "react-router";
+import "../Explorer/ExplorerPage.css";
+import { useNavigate } from "react-router";
+import { FadeLoader } from "react-spinners";
+import ExplorerService from "../../services/ExplorerService";
+import DestinationSection from "../../DestinationSection/DestinationSection";
+
+export default function DetailsPage() {
+  const [weather, setWeather] = useState({});
+  const { place } = useParams();
+  const [placeData, setData] = useState([]);
+  const [relatedPlace, setRelatedPlaces] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const weatherData = await ExplorerService.getWeather(place);
+        setWeather(weatherData);
+
+        const placeData = await ExplorerService.getPlaceData(place);
+        setData(placeData);
+
+        // Fetch related places
+        if (placeData.relatedPlaces) {
+          const relatedPlacesData = await ExplorerService.getRelatedPlaces(
+            placeData.relatedPlaces
+          );
+          setRelatedPlaces(relatedPlacesData);
+          setIsLoading(false);
+        }
+      } catch (error) {
+        navigate("/home");
+      }
+    };
+
+    fetchData();
+  }, [place, navigate]);
+
+  // To split the para content
+  const placeDescription = placeData.fullDescription
```

Comment:
could you explain why the split is made here

Reply:
Its because in Details Page there is a spilt of paragraph to split based on \n to new line